src/createProjector.ts

```diff
@@ -0,0 +1,187 @@
+import { ComposeFactory } from 'dojo-compose/compose';
+import { Handle } from 'dojo-interfaces/core';
+import { VNodeProperties } from 'dojo-interfaces/vdom';
+import { Widget, WidgetState, WidgetOptions } from 'dojo-interfaces/widgetBases';
+import WeakMap from 'dojo-shim/WeakMap';
+import { createProjector as createMaquetteProjector, Projector as MaquetteProjector } from 'maquette';
+import createWidgetBase from './bases/createWidgetBase';
+import global from 'dojo-core/global';
+import Promise from 'dojo-shim/Promise';
+
+/**
+ * Represents the state of the projector
+ */
+export enum ProjectorState {
+	Attached = 1,
+	Detached
+};
+
+/**
+ * Projector interface
+ */
+export interface ProjectorOptions extends WidgetOptions<WidgetState> {
+
+	/**
+	 * An optional root of the projector
+	 */
+	root?: Element;
+
+	/**
+	 * If `true`, will configure the projector to support css transitions using `cssTransitions` global object.
+	 * The projector will fail create if the options is true but the global object cannot be found.
+	 */
+	cssTransitions?: boolean;
+}
+
+export interface ProjectorMixin {
+
+	/**
+	 * Attach the projector to the an element provided as the root.
+	 */
+	attach(): Promise<Handle>;
+
+	/**
+	 * Root element to attach the projector
+	 */
+	root: Element;
+
+	/**
+	 * The Maquette projector
+	 */
+	readonly projector: MaquetteProjector;
+
+	/**
+	 * The status of the projector
+	 */
+	readonly state: ProjectorState;
+}
+
+/**
+ * Internal projector state
+ */
+interface ProjectorData {
+	projector: MaquetteProjector;
+	root: Element;
+	state: ProjectorState;
+	attachPromise?: Promise<Handle>;
+	attachHandle?: Handle;
+	afterCreate?: () => void;
+}
+
+export type Projector = Widget<WidgetState> & ProjectorMixin;
```

Comment:
That would be good, but I cannot get this to work 😢

```ts
export interface ProjectorOverloads {
    on(type: 'render:scheduled', listener: EventListener<Projector, EventTargettedObject<Projector>>);
    on(type: 'projector:attached', listener: EventListener<Projector, EventTargettedObject<Projector>>);
}
```

and added to `export type Projector = Widget<WidgetState> & ProjectorMixin & ProjectorOverloads;` but got an error that included `Types of property 'mixin' are incompatible.`. 

not sure what I am doing wrong.